addons/hr/models/hr_employee.py

```diff
@@ -241,6 +241,14 @@ def get_view(self, view_id=None, view_type='form', **options):
             return super().get_view(view_id, view_type, **options)
         return self.env['hr.employee.public'].get_view(view_id, view_type, **options)
 
+    @api.model
+    def get_views(self, views, options):
+        if self.check_access_rights('read', raise_exception=False):
+            return super(HrEmployeePrivate, self).get_views(views, options)
+        res = self.env['hr.employee.public'].get_views(views, options)
+        res['models'].update({'hr.employee': res['models']['hr.employee.public']})
```

Comment:
I am not sure about it :thinking: It's strange that we have `hr.employee` as key and give it view for `hr.employee.public`?

I don't know how dashboard is working, but can we not do the same as in other views? Can we not ask specifically which model to search?

Reply:
It's because in other cases, we use the relation option directly on the view. This works as long as we have access to the view. However, dashboards are different since users are able to create their own dashboards. That means that we have no possibility from the front end to specify what to do in which case from there :thinking: This is indeed a bit hacky, but I'm not sure we can do it in another way :sweat_smile: